src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -252,18 +252,39 @@ private IEnumerable<DiagnosticAnalyzer> GetAnalyzersForTypeNames(Assembly analyz
             // Given the type names, get the actual System.Type and try to create an instance of the type through reflection.
             foreach (var typeName in analyzerTypeNames)
             {
+                Type type;
+                string loadErrorMessage = CodeAnalysisResources.TypeOrDependencyNotFound;
```

Comment:
If I'm reading the code correctly, the initial value is overwritten on all code paths, if that is so, either the initial value can be elided or it can be made null (and one of the assignments below removed).